triedb/database.go

```diff
@@ -119,13 +121,16 @@ func NewDatabase(diskdb ethdb.Database, config *Config) *Database {
 		}
 		db.backend = hashdb.New(diskdb, config.HashDB, resolver)
 	}
+	db.overrideBackend(diskdb, config)
```

Comment:
if we handle this here then if `config.IsVerkle` is set the hashDB path will try to initialize but fail

Reply:
this is an example (non-recommended) fix https://github.com/ethereum/go-ethereum/commit/753d1f5dff5f607182c5fc32d09c601f62acdaa7 but probably a different order of checking whether the optional config is specified

Reply:
Nice catch, thank you. I was too focussed on mechanistic rules for where changes are made (in this case, "last thing before the return") so we can one day define how to resolve conflicts.

I've moved it to immediately after `db` is declared and the override now returns a bool that signals an early return. This also removes the smell of having to close a just-opened backend.